.changeset/wet-jeans-repair.md

```diff
@@ -0,0 +1,5 @@
+---
+"@biomejs/biome": patch
+---
+
+Fix #6675: Only flag noAccumulatingSpread on Object.assign when a new object is being allocated on each iteration. Ex: foo.reduce((acc, bar) => Object.assign({}, acc, bar), {})
```

Comment:
The contribution guide explains how to write the changeset (the PR template has a link to that paragraph):
- sentences should start with the past tense
- link the issue 
- examples should be inside a code block

Refers to other changesets as examples